tests/js-custom-authorization/custom_authorization.py

```diff
@@ -356,6 +365,339 @@ def test_jwt_auth(network, args):
         assert r.status_code == HTTPStatus.UNAUTHORIZED, r.status_code
 
+    network.consortium.remove_jwt_issuer(primary, issuer.name)
+    return network
+
+
+def try_auth(primary, issuer, kid, iss, tid):
+    with primary.client("user0") as c:
+        LOG.info("Calling JWT with kid from issuer for tenant")
+        r = c.get(
+            "/app/jwt",
+            headers=infra.jwt_issuer.make_bearer_header(
+                issuer.issue_jwt(kid, claims={"iss": iss, "tid": tid})
+            ),
+        )
+        assert r.status_code
+        return r.status_code
+
+
+@reqs.description("JWT authentication as by MSFT Entra (single tenant)")
+def test_jwt_auth_msft_single_tenant(network, args):
+    """For a specific tenant, only tokens with this issuer+tenant can auth."""
+
+    primary, _ = network.find_nodes()
+
+    TENANT_ID = "9188050d-6c67-4c5b-b112-36a304b66da"
+    ISSUER_TENANT = (
+        "https://login.microsoftonline.com/9188050d-6c67-4c5b-b112-36a304b66da/v2.0"
+    )
+
+    issuer = infra.jwt_issuer.JwtIssuer(name="https://login.microsoftonline.com")
+    jwt_kid = "my_key_id"
+
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as metadata_fp:
+        jwt_cert_der = infra.crypto.cert_pem_to_der(issuer.cert_pem)
+        der_b64 = base64.b64encode(jwt_cert_der).decode("ascii")
+        data = {
+            "issuer": issuer.name,
+            "auto_refresh": False,
+            "jwks": {
+                "keys": [
+                    {
+                        "kty": "RSA",
+                        "kid": jwt_kid,
+                        "x5c": [der_b64],
+                        "issuer": ISSUER_TENANT,
+                    }
+                ]
+            },
+        }
+        json.dump(data, metadata_fp)
+        metadata_fp.flush()
+        network.consortium.set_jwt_issuer(primary, metadata_fp.name)
+
+    assert (
+        try_auth(primary, issuer, jwt_kid, ISSUER_TENANT, "garbage_tenant")
+        == HTTPStatus.UNAUTHORIZED
+    )
+    assert (
+        try_auth(primary, issuer, jwt_kid, ISSUER_TENANT, "{tenantid}")
+        == HTTPStatus.UNAUTHORIZED
+    )
+    assert try_auth(primary, issuer, jwt_kid, ISSUER_TENANT, TENANT_ID) == HTTPStatus.OK
+
+    network.consortium.remove_jwt_issuer(primary, issuer.name)
+
+    assert (
+        try_auth(primary, issuer, jwt_kid, ISSUER_TENANT, TENANT_ID)
+        == HTTPStatus.UNAUTHORIZED
+    )
+
+    return network
+
+
+@reqs.description("JWT authentication as by MSFT Entra (multiple tenants)")
+def test_jwt_auth_msft_multitenancy(network, args):
+    """For a common tenant, all tokens from this issuer can auth,
+    no matter which tenant is specified."""
+
+    primary, _ = network.find_nodes()
+
+    COMMNON_ISSUER = "https://login.microsoftonline.com/{tenantid}/v2.0"
+    TENANT_ID = "9188050d-6c67-4c5b-b112-36a304b66da"
+    ISSUER_TENANT = f"https://login.microsoftonline.com/{TENANT_ID}/v2.0"
+    ANOTHER_TENANT_ID = "ANOTHER-6c67-4c5b-b112-36a304b66da"
```

Comment:
```suggestion
    ANOTHER_TENANT_ID = "aaaaaaa-6c67-4c5b-b112-36a304b66da"
```

Nit: Although we treat these as strings for now, they're really GUIDs - we should use a string of the right shape so that it will still work if we add more validation later. Hopefully it's clear that this is still an invalid/constructed GUID.